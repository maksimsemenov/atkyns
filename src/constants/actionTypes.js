export const CHANGE_FIELD = 'CHANGE_FIELD'
export const ADD_CASE = 'ADD_CASE'

export const SET_STAGE = 'SET_STAGE'

export const SET_PAYMENT = 'SET_PAYMENT'
