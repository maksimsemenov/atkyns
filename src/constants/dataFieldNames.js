export const RELASHIONSHIP = 'relashionship'
export const REL_BY_ADOPTION = 'relByAdoption'
export const RES_THROUGH_ADOPTION = 'resThroughAdoption'

export const P_FIRST_NAME = 'petitioner/name/first'
export const P_MIDDLE_NAME = 'petitioner/name/middle'
export const P_FAMILY_NAME = 'petitioner/name/family'
export const P_OTHER_NAME = 'petitioner/name/other'
export const P_ADDRESS_STREET = 'petitioner/address/street'
export const P_ADDRESS_APT_NUMBER = 'petitioner/address/apartment'
export const P_ADDRESS_CITY = 'petitioner/address/city'
export const P_ADDRESS_STATE = 'petitioner/address/state'
export const P_ADDRESS_COUNTRY = 'petitioner/address/country'
export const P_ADDRESS_ZIP = 'petitioner/address/zip'
export const P_BIRTH_CITY = 'petitioner/BirthCity'
export const P_BIRTH_STATE = 'petitioner/BirthState'
export const P_BIRTH_COUNTRY = 'petitioner/BirthCountry'
export const P_BIRTH_DATE = 'petitioner/BirthDate'
export const P_GENDER = 'petitioner/Gender'
export const P_MARITAL_STATUS = 'petitioner/maritalStatus'
export const P_PMARRIAGE_DATE = 'petitioner/PMarriageDate'
export const P_PMARRIAGE_CITY = 'petitioner/PMarriageCity'
export const P_PMARRIAGE_STATE = 'petitioner/PMarriageState'
export const P_PMARRIAGE_COUNTRY = 'petitioner/PMarriageCountry'
export const P_SSN  ='petitioner/SSN'
export const P_ARN = 'petitioner/ARN'
export const P_PRIOR_MARRIAGES = 'petitioner/PriorMarriages'
export const P_STATUS_TYPE = 'petitioner/status/type'

export const R_FIRST_NAME = 'relative/name/first'
export const R_MIDDLE_NAME = 'relative/name/middle'
export const R_FAMILY_NAME = 'relative/name/family'
export const R_OTHER_NAME = 'relative/name/other'
export const R_NATIVE_NAME = 'relative/name/native'
