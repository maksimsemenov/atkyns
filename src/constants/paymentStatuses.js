export const PAYMENT_NONE = 'PAYMENT_NONE'
export const PAYMENT_FULL = 'PAYMENT_FULL'
