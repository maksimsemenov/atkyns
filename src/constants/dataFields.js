/* @flow */
/* This file describes all data fields that we have in a project as well as relationships between them. Relationsips describes as effects that one field will cause on another
*/

import type { DataFieldT } from '../types/DataFieldT'

import * as fieldTypes from 'constants/fieldTypes'
import * as relTypes from 'constants/relationshipTypes'
import { STATUS_TYPE_CITIZEN, STATUS_TYPE_PERMANENT } from 'constants/default'
import * as DFN from 'constants/dataFieldNames'

const dataFields = new Map({
  [DFN.P_STATUS_TYPE]: {
    name: DFN.P_STATUS_TYPE,
    type: fieldTypes.OPTION,
    options: [ STATUS_TYPE_CITIZEN, STATUS_TYPE_PERMANENT ]
  },
  [DFN.RELASHIONSHIP]: {
    name: DFN.RELASHIONSHIP,
    value: '',
    type: fieldTypes.OPTION,
    disable: false,
    options: [
      { value: relTypes.SPOUSE },
      { value: relTypes.PARENT },
      { value: relTypes.BROTHER },
      { value: relTypes.CHILD },
    ],
    effects: [
      {
        fieldName: DFN.REL_BY_ADOPTION,
        effect: ({ value }) => ({
          disable: value === relTypes.SPOUSE ? true : false
        })
      }
    ]
  },

})[
  {
    name: DFN.RELASHIONSHIP,
    value: '',
    type: fieldTypes.OPTION,
    disable: false,
    options: [
      { value: relTypes.SPOUSE },
      { value: relTypes.PARENT },
      { value: relTypes.BROTHER },
      { value: relTypes.CHILD },
    ],
    effects: [
      {
        fieldName: DFN.REL_BY_ADOPTION,
        effect: ({ value }) => ({
          disable: value === relTypes.SPOUSE ? true : false
        })
      }
    ]
  },
  {
    name: 'relByAdoption',
    value: '',
    type: fieldTypes.BOOLEAN,
    disable: false
  },
  {
    name: 'resThroughAdoption',
    value: '',
    type: fieldTypes.BOOLEAN,
    disable: false
  },
  {
    name: 'pfirstName',
    value: '',
    type: fieldTypes.TEXT,
    disable: false
  }
]

export const petitionerStatus:

export default dataFields
