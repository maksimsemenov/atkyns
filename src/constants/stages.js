import * as fieldNames from 'constants/dataFieldNames'

export const STAGE_OVERVIEW = 'overview'
export const STAGE_PETITIONER = 'petitioner'
export const STAGE_RELATIVE = 'relative'
export const STAGE_PETITION = 'petition'

export const stages = [
  { name: STAGE_OVERVIEW },
  { name: STAGE_PETITIONER },
  { name: STAGE_RELATIVE },
  { name: STAGE_PETITION }
]

// const stages = {
//   '0': [
//     fieldNames.RELASHIONSHIP,
//     fieldNames.REL_BY_ADOPTION,
//     fieldNames.RES_THROUGH_ADOPTION
//   ]
// }
//
// export default stages
