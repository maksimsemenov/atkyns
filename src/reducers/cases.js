import { fromJS } from 'immutable'
import capitalize from 'lodash/capitalize'
import trim from 'lodash/trim'
import l from 'utils/local'
import caseReducer, * as fromCase from './case'
import * as DFN from 'constants/dataFieldNames'
import { CHANGE_FIELD, ADD_CASE, SET_STAGE, SET_PAYMENT, DELETE_CASE, RESTORE_CASE } from 'constants/actionTypes'
import { PAYMENT_FULL } from 'constants/paymentStatuses'

const testInitialState = {
  '415rd1wei6o': {
    stage: 0,
    payments: PAYMENT_FULL,
    data: {
      [DFN.P_FIRST_NAME]: {
        value: 'Andy'
      },
      [DFN.P_FAMILY_NAME]: {
        value: 'Smith'
      },
      [DFN.R_FIRST_NAME]: {
        value: 'Veronica'
      },
      [DFN.R_FAMILY_NAME]: {
        value: 'Landau'
      },
      [DFN.P_GENDER]: {
        value: ''
      },
      [DFN.P_CITY]: {
        value: ''
      }
    }
  }
}

const cases = (state = fromJS({}), action) => {
  switch (action.type) {
    case ADD_CASE:
      return state.set(action.caseId, caseReducer(undefined, action))
    case CHANGE_FIELD:
    case SET_STAGE:
    case SET_PAYMENT:
      const { caseId } = action
      return state.set(caseId, caseReducer(state.get(caseId, fromJS({})), action))
    case DELETE_CASE:
      return state.setIn([action.caseId, 'deleted'], true)
    case RESTORE_CASE:
      return state.setIn([action.caseId, 'deleted'], false)
    default:
      return state
  }
}

export default cases


/*
 * Selectors
 */

const caseFields = [
  DFN.P_FIRST_NAME,
  DFN.P_FAMILY_NAME,
  DFN.R_FIRST_NAME,
  DFN.R_FAMILY_NAME
]

export const getCaseIdsList = (state = fromJS({})) => state.keySeq().toJS()
export const getCasesNumber = (state = fromJS({})) => state.size
export const getCaseStage = (state = fromJS({}), caseId) => fromCase.getStage(state.get(caseId))
export const getCaseProgress = (state = fromJS({}), caseId) => fromCase.getProgress(state.get(caseId))
export const getCasePaymentStatuse = (state = fromJS({}), caseId) => fromCase.getPaymentStatuse(state.get(caseId))
export const getCaseData = (state = fromJS({}), caseId) => fromCase.getData(state.get(caseId))
export const getCaseDataField = (state = fromJS({}), caseId, fieldName) => fromCase.getDataField(state.get(caseId), fieldName)
export const getCaseDataFields = (state = fromJS({}), caseId, fields) => fromCase.getDataFields(state.get(caseId), fields)

export const getCasesList = (state = fromJS({})) => {
  return getCaseIdsList(state).map((caseId, index) => {
    const caseData = getCaseDataFields(state, caseId, caseFields)
    const pName = caseData[DFN.P_FIRST_NAME] || caseData[DFN.P_FAMILY_NAME] ?
      trim(`${capitalize(caseData[DFN.P_FIRST_NAME])} ${capitalize(caseData[DFN.P_FAMILY_NAME])}`) :
      `${l('%case')} ${index + 1}`
    const rName = trim(`${capitalize(caseData[DFN.R_FIRST_NAME])} ${capitalize(caseData[DFN.R_FAMILY_NAME])}`)

    return {
      id: caseId,
      pName: pName,
      rName: rName,
      stage: getCaseStage(state, caseId),
      progress: getCaseProgress(state, caseId),
  }})
}
