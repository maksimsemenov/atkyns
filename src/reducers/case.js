/* @flow */
import type { ActionT } from 'types/ActionT'

import { fromJS, List, Map } from 'immutable'
// import values from 'lodash/values'
import { actionTypes } from 'reducers/cases'
import { STAGE_OVERVIEW } from 'constants/stages'
import { PAYMENT_NONE } from 'constants/paymentStatuses'
// import * as DFN from 'constants/dataFieldNames'
//
// export const initialData = (fields: string[]) => (
//   fields.reduce((newData, fieldName) => {
//     newData[fieldName] = { value: '', disable: false }
//     return newData
//   }, {})
// )
export const initialState = () => fromJS({
  stage: STAGE_OVERVIEW,
  payment: PAYMENT_NONE,
  data: {}
})

const caseReducer = (state: Map<string, any> = initialState(), action: ActionT) => {
  switch (action.type) {
    case actionTypes.CHANGE_FIELD:
      const { fieldName = '', value } = action
      const path = fieldName.split('.')
      return state.setIn(['data', ...path], value)
    case actionTypes.SET_STAGE:
      return state.set('stage', action.newStage || STAGE_OVERVIEW)
    case actionTypes.SET_PAYMENT:
      return state.set('payment', action.newPaymentStatuse || PAYMENT_NONE)
    case actionTypes.DELETE_CASE:
      return state.set('deleted', true)
    case actionTypes.RESTORE_CASE:
      return state.set('deleted', false)
    case actionTypes.SET_FIELD_ERROR:
      return state.setIn(['errors', action.fieldName], action.error)
    case actionTypes.REMOVE_FIELD_ERROR:
      return state.deleteIn(['errors', action.fieldName])
    default:
      return state
  }
}

export default caseReducer

/*
 * Selectors
 */

export const getStage = (state: Map<string, any> = fromJS({})) => state.get('stage', 0)
export const getPaymentStatuse = (state: Map<string, any> = fromJS({})) => state.get('payment', PAYMENT_NONE)
export const getData = (state: Map<string, any> = fromJS({})) => state.get('data', fromJS({})).toJS()
export const getDataField = (state: Map<string, any> = fromJS({}), fieldName: string = '') => {
  const path = fieldName.split('.')
  const field = state.getIn(['data', ...path], '')
  return List.isList(field) || Map.isMap(field) ?
    field.toJS() :
    field
}
export const getDataFields = (state: Map<string, any> = fromJS({}), fields: string[] = []) => (
  fields.reduce((fMap, fieldName) => {
    fMap[fieldName] = getDataField(state, fieldName)
    return fMap
  }, {})
)
export const getDataFieldError = (
  state: Map<string, any> = fromJS({}),
  fieldName: string = ''
) => state.getIn(['errors', fieldName])
