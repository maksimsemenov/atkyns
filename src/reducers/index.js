import { fromJS } from 'immutable'
import cases, * as fromCases from 'reducers/cases'

export const combineReducers = (reducers) => (state = fromJS({}), action = {}) => (
  Object.keys(reducers).reduce((s, r) => (
    s.set(r, reducers[r](state.get(r), action))
  ), state)
)

const rootReducer = combineReducers({ cases })

export default rootReducer

/*
 * Selectors
 */

export const getCasesNumber = (state = fromJS({})) => fromCases.getCasesNumber(state.get('cases'))
export const getActiveCasesNumber = (state = fromJS({})) => fromCases.getActiveCasesNumber(state.get('cases'))
export const getDeletedCasesNumber = (state = fromJS({})) => fromCases.getDeletedCasesNumber(state.get('cases'))
export const getCaseStage = (state = fromJS({}), caseId) => fromCases.getCaseStage(state.get('cases'), caseId)
export const getCaseProgress = (state = fromJS({}), caseId) => fromCases.getCaseProgress(state.get('cases'), caseId)
export const getCasePaymentStatuse = (state = fromJS({}), caseId) => fromCases.getCasePaymentStatuse(state.get('cases'), caseId)
export const getCaseData = (state = fromJS({}), caseId) => fromCases.getCaseData(state.get('cases'), caseId)
export const getCaseDataField = (state = fromJS({}), caseId, fieldName) => fromCases.getCaseDataField(state.get('cases'), caseId, fieldName)
export const getCaseDataFields = (state = fromJS({}), caseId, fields) => fromCases.getCaseDataFields(state.get('cases'), caseId, fields)
export const getCasesList = (state = fromJS({}), deleted = false) => fromCases.getCasesList(state.get('cases'), deleted)
