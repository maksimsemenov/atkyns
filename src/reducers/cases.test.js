import { fromJS } from 'immutable'
import casesReducer, * as fromCases from 'reducers/cases'
import { PAYMENT_NONE, PAYMENT_FULL } from 'constants/paymentStatuses'
import { ADD_CASE, SET_STAGE, SET_PAYMENT, DELETE_CASE, RESTORE_CASE } from 'reducers/cases'
import { STAGE_OVERVIEW, STAGE_PETITIONER } from 'constants/stages'

describe('Cases reducer', () => {
  it('handles ADD_CASE action', () => {
    const nextState = fromJS({
      jlfdsfsd: {
        stage: STAGE_OVERVIEW,
        payment: PAYMENT_NONE,
        data: {}
      }
    })
    const action = { type: ADD_CASE, caseId: 'jlfdsfsd' }
    expect(casesReducer(undefined, action)).toEqual(nextState)
  })
  it('handle SET_STAGE action', () => {
    const state = fromJS({
      fdfdfa: { stage: 4 },
      roqwsa: { stage: 7 }
    })
    const action = {
      type: SET_STAGE,
      caseId: 'fdfdfa',
      newStage: 8
    }
    const nextState = fromJS({
      fdfdfa: { stage: 8 },
      roqwsa: { stage: 7 }
    })
    expect(casesReducer(state, action)).toEqual(nextState)
  })
  it('handle SET_PAYMENT action', () => {
    const state = fromJS({
      fdfdfa: { payment: PAYMENT_NONE },
      roqwsa: { payment: PAYMENT_FULL }
    })
    const action = {
      type: SET_PAYMENT,
      caseId: 'fdfdfa',
      newPaymentStatuse: PAYMENT_FULL
    }
    const nextState = fromJS({
      fdfdfa: { payment: PAYMENT_FULL },
      roqwsa: { payment: PAYMENT_FULL }
    })
    expect(casesReducer(state, action)).toEqual(nextState)
  })
  it('handle DELETE_CASE action', () => {
    const state = fromJS({
      jlfdsfsd: { deleted: false },
      lkdeweqw: { deleted: true },
      daslksal: {}
    })
    const action1 = { type: DELETE_CASE, caseId: 'jlfdsfsd' }
    const action2 = { type: DELETE_CASE, caseId: 'lkdeweqw' }
    const action3 = { type: DELETE_CASE, caseId: 'daslksal' }

    const nextState1 = fromJS({
      jlfdsfsd: { deleted: true },
      lkdeweqw: { deleted: true },
      daslksal: {}
    })
    const nextState2 = fromJS({
      jlfdsfsd: { deleted: false },
      lkdeweqw: { deleted: true },
      daslksal: {}
    })
    const nextState3 = fromJS({
      jlfdsfsd: { deleted: false },
      lkdeweqw: { deleted: true },
      daslksal: { deleted: true }
    })

    expect(casesReducer(state, action1)).toEqual(nextState1)
    expect(casesReducer(state, action2)).toEqual(nextState2)
    expect(casesReducer(state, action3)).toEqual(nextState3)
  })
  it('handle RESTORE_CASE action', () => {
    const state = fromJS({
      jlfdsfsd: { deleted: false },
      lkdeweqw: { deleted: true },
      daslksal: {}
    })
    const action1 = { type: RESTORE_CASE, caseId: 'jlfdsfsd' }
    const action2 = { type: RESTORE_CASE, caseId: 'lkdeweqw' }
    const action3 = { type: RESTORE_CASE, caseId: 'daslksal' }

    const nextState1 = fromJS({
      jlfdsfsd: { deleted: false },
      lkdeweqw: { deleted: true },
      daslksal: {}
    })
    const nextState2 = fromJS({
      jlfdsfsd: { deleted: false },
      lkdeweqw: { deleted: false },
      daslksal: {}
    })
    const nextState3 = fromJS({
      jlfdsfsd: { deleted: false },
      lkdeweqw: { deleted: true },
      daslksal: { deleted: false }
    })

    expect(casesReducer(state, action1)).toEqual(nextState1)
    expect(casesReducer(state, action2)).toEqual(nextState2)
    expect(casesReducer(state, action3)).toEqual(nextState3)
  })
})
it('getCaseIdsList selector returns correct list', () => {
  const state = fromJS({
    dfdafad: { name: 'Andy' },
    fdffewc: { name: 'Victoria'}
  })
  expect(fromCases.getCaseIdsList(state)).toEqual(['dfdafad', 'fdffewc'])
})
it('getActiveCaseIdsList selector returns correct list', () => {
  const state = fromJS({
    dfdafad: { data: { name: 'Andy' }, deleted: false },
    fdffewc: { data: { name: 'Victoria' }, deleted: true },
    ksaldll: { data: { name: 'Many' }, deleted: false },
  })
  expect(fromCases.getActiveCaseIdsList(state)).toEqual(['dfdafad', 'ksaldll'])
})
it('getDeletedCaseIdsList selector returns correct list', () => {
  const state = fromJS({
    dfdafad: { data: { name: 'Andy' }, deleted: true },
    fdffewc: { data: { name: 'Victoria' }, deleted: true },
    ksaldll: { data: { name: 'Many' }, deleted: false },
  })
  expect(fromCases.getDeletedCaseIdsList(state)).toEqual(['dfdafad', 'fdffewc'])
})
it('getCasesNumber', () => {
  const state = fromJS({
    dfdafad: { name: 'Andy' },
    fdffewc: { name: 'Victoria'}
  })
  expect(fromCases.getCasesNumber(state)).toBe(2)
  expect(fromCases.getCasesNumber(undefined)).toBe(0)
})
it('getActiveCasesNumber', () => {
  const state = fromJS({
    dfdafad: { name: 'Andy' },
    fdffewc: { name: 'Victoria'},
    lkdjald: { name: 'Jack', deleted: true}
  })
  expect(fromCases.getActiveCasesNumber(state)).toBe(2)
  expect(fromCases.getActiveCasesNumber(undefined)).toBe(0)
})
it('getDeletedCasesNumber', () => {
  const state = fromJS({
    dfdafad: { name: 'Andy' },
    fdffewc: { name: 'Victoria'},
    lkdjald: { name: 'Jack', deleted: true}
  })
  expect(fromCases.getDeletedCasesNumber(state)).toBe(1)
  expect(fromCases.getDeletedCasesNumber(undefined)).toBe(0)
})
it('getCaseStage', () => {
  const state = fromJS({
    dfdafad: { stage: STAGE_OVERVIEW },
    fdffewc: { stage: STAGE_PETITIONER }
  })
  expect(fromCases.getCaseStage(state, 'dfdafad')).toBe(STAGE_OVERVIEW)
})
it('getCasePaymentStatuse', () => {
  const state = fromJS({
    dfdafad: { payment: PAYMENT_NONE },
    fdffewc: { payment: PAYMENT_FULL }
  })
  expect(fromCases.getCasePaymentStatuse(state, 'dfdafad')).toBe(PAYMENT_NONE)
  expect(fromCases.getCasePaymentStatuse(state, 'fdffewc')).toBe(PAYMENT_FULL)
})
it('getCaseData', () => {
  const state = fromJS({
    dfdafad: {
      data: {
        firstName: {
          value: 'Andy'
        },
        familyName: {
          value: ''
        },
        middleName: {
          disable: true
        }
      }
    },
    fdffewc: {
      data: {
        firstName: {
          value: 'Andy'
        },
        familyName: {
          value: 'Smith'
        },
      }
    }
  })
  const data1 = {
    firstName: {
      value: 'Andy'
    },
    familyName: {
      value: ''
    },
    middleName: {
      disable: true
    }
  }
  const data2 = {
    firstName: {
      value: 'Andy'
    },
    familyName: {
      value: 'Smith'
    },
  }
  expect(fromCases.getCaseData(state, 'dfdafad')).toEqual(data1)
  expect(fromCases.getCaseData(state, 'fdffewc')).toEqual(data2)
})
it('getCaseDataField', () => {
  const state = fromJS({
    dfdafad: {
      data: {
        petitioner: {
          name: {
            first: 'Andy',
            family: '',
            middle: 'Largo'
          }
        }
      }
    },
    fdffewc: {
      data: {
        relative: {
          name: {
            first: 'Olef',
            family: 'Gilbert'
          }
        }
      }
    }
  })
  expect(fromCases.getCaseDataField(state, 'dfdafad', 'petitioner.name.first')).toEqual('Andy')
  expect(fromCases.getCaseDataField(state, 'fdffewc', 'relative.name.family')).toEqual('Gilbert')
})
it('getCaseDataFields', () => {
  const state = fromJS({
    dfdafad: {
      data: {
        petitioner: {
          name: {
            first: 'Andy',
            family: '',
            middle: 'Largo'
          }
        },
        relative: {
          name: {
            first: 'Olef',
            family: 'Gilbert'
          }
        }
      }
    },
    fdffewc: {
      data: {
        petitioner: {
          name: {
            first: 'Andy',
            family: '',
            middle: 'Largo'
          }
        },
        relative: {
          name: {
            first: 'Olef',
            family: 'Gilbert'
          }
        }
      }
    }
  })
  const fields = ['petitioner.name.first', 'relative.name.family']
  const data = {
    'petitioner.name.first': 'Andy',
    'relative.name.family': 'Gilbert'
  }
  expect(fromCases.getCaseDataFields(state, 'dfdafad', fields)).toEqual(data)
})
it('getCasesList', () => {
  const state = fromJS({
    dkdlasdad: {
      stage: STAGE_PETITIONER,
      data: {
        petitioner: {
          name: {
            first: 'Andy',
            family: 'Smith'
          }
        },
        relative: {
          name: {
            first: 'Victoria',
            family: ''
          }
        }
      }
    },
    dkopesdad: {
      stage: STAGE_OVERVIEW,
      data: {}
    },
    orpwesdad: {
      stage: STAGE_OVERVIEW,
      deleted: true,
      data: {
        petitioner: {
          name: {
            first: 'Sergey',
            family: 'Lohonev'
          }
        }
      }
    }
  })
  const activeCasesList = [
    {
      id: 'dkdlasdad',
      pName: 'Andy Smith',
      rName: 'Victoria',
      stage: STAGE_PETITIONER
    },
    {
      id: 'dkopesdad',
      pName: '%case 2',
      rName: '',
      stage: STAGE_OVERVIEW
    }
  ]
  const deletedCasesList = [
    {
      id: 'orpwesdad',
      pName: 'Sergey Lohonev',
      rName: '',
      stage: STAGE_OVERVIEW
    }
  ]
  expect(fromCases.getCasesList(state)).toEqual(activeCasesList)
  expect(fromCases.getCasesList(state, true)).toEqual(deletedCasesList)
})
it ('getDataFieldError selector returns correct value', () => {
  const state = fromJS({
    dasdasfd: {
      errors: {
        'petitioner.name.first': '%error-emptyField'
      }
    },
    dsadados: {
      errors: {
        'petitioner.name.second': undefined
      }
    }
  })
  expect(fromCases.getCaseDataFieldError(state, 'dasdasfd', 'petitioner.name.first'))
    .toEqual('%error-emptyField')
  expect(fromCases.getCaseDataFieldError(state, 'dsadados', 'petitioner.name.second'))
    .toBeUndefined()
})

/*
 * Action creators
 */

it('emits events for action effects', () => {
 let register = {}
 const dispatch = (action) => {
   if (typeof action === 'function') {
     action(dispatch)
   } else {
     register.caseId = action.caseId
     register[action.fieldName] = action.value
   }
 }
 const effects = [{
   fieldName: 'secondName',
   effect: () => 'Smith'
 }]
 const nextRegister = {
   caseId: '0f',
   firstName: 'Andy',
   secondName: 'Smith',
 }
 dispatch(fromCases.actions.changeDataField('0f', 'firstName', 'Andy', effects))
 expect(register).toEqual(nextRegister)
})

it('setFieldError action creator', () => {
  expect(fromCases.actions.setFieldError('dsadhaskld', 'petitioner.name.first', '%error-emptyField')).toMatchSnapshot()
})
it('removeFieldError action creator', () => {
  expect(fromCases.actions.removeFieldError('dsadhaskld', 'petitioner.name.first')).toMatchSnapshot()
})
it('addCase action creator', () => {
  expect(fromCases.actions.addCase('dsadhaskld')).toMatchSnapshot()
})
it('deleteCase action creator', () => {
  expect(fromCases.actions.deleteCase('dsadhaskld')).toMatchSnapshot()
})
it('restoreCase action creator', () => {
  expect(fromCases.actions.restoreCase('dsadhaskld')).toMatchSnapshot()
})
it('restoreCase action creator', () => {
  expect(fromCases.actions.restoreCase('dsadhaskld')).toMatchSnapshot()
})
it('setStage action creator', () => {
  expect(fromCases.actions.setStage('dsadhaskld', 'test-stage')).toMatchSnapshot()
})
it('setPayment action creator', () => {
  expect(fromCases.actions.setPayment('dsadhaskld', 'test-payment-status')).toMatchSnapshot()
})
