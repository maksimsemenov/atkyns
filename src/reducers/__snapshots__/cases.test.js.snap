exports[`test addCase action creator 1`] = `
Object {
  "caseId": "dsadhaskld",
  "type": "ADD_CASE"
}
`;

exports[`test deleteCase action creator 1`] = `
Object {
  "caseId": "dsadhaskld",
  "type": "DELETE_CASE"
}
`;

exports[`test removeFieldError action creator 1`] = `
Object {
  "caseId": "dsadhaskld",
  "fieldName": "petitioner.name.first",
  "type": "REMOVE_FIELD_ERROR"
}
`;

exports[`test restoreCase action creator 1`] = `
Object {
  "caseId": "dsadhaskld",
  "type": "RESTORE_CASE"
}
`;

exports[`test setFieldError action creator 1`] = `
Object {
  "caseId": "dsadhaskld",
  "error": "%error-emptyField",
  "fieldName": "petitioner.name.first",
  "type": "SET_FIELD_ERROR"
}
`;

exports[`test setPayment action creator 1`] = `
Object {
  "caseId": "dsadhaskld",
  "newPaymentStatuse": "test-payment-status",
  "type": "SET_PAYMENT"
}
`;

exports[`test setStage action creator 1`] = `
Object {
  "caseId": "dsadhaskld",
  "newStage": "test-stage",
  "type": "SET_STAGE"
}
`;
